mlrun/runtimes/nuclio/application/application.py

```diff
@@ -344,24 +335,31 @@ def deploy(
             builder_env=builder_env,
         )
         logger.info(
-            "Successfully deployed function, creating API gateway",
-            api_gateway_name=self.status.api_gateway_name,
-            authentication_mode=authentication_mode,
+            "Successfully deployed function.",
         )
 
         # Restore the source in case it was removed to make nuclio not consider it when building
         if not self.spec.build.source and self.status.application_source:
             self.spec.build.source = self.status.application_source
         self.save(versioned=False)
 
-        ports = self.spec.internal_application_port if direct_port_access else []
-        return self.create_api_gateway(
-            name=self.status.api_gateway_name,
-            ports=ports,
-            authentication_mode=authentication_mode,
-            authentication_creds=authentication_creds,
-            ssl_redirect=ssl_redirect,
-        )
+        if create_default_api_gateway:
+            try:
+                api_gateway_name = self._resolve_default_api_gateway_name()
+                return self.create_api_gateway(api_gateway_name, set_as_default=True)
+            except Exception as exc:
+                logger.warning(
+                    "Failed to create default API gateway, application will not be accessible. "
+                    "Use the `create_api_gateway` method to make it accessible",
+                    exc=mlrun.errors.err_to_str(exc),
+                )
+        else:
+            logger.warning(
+                "Application is online but is not accessible. "
```

Comment:
```suggestion
                "Application is online but may not be accessible since default gateway creation was not requested."
```
It's still a user's choice to get here - so make sure it's understood this is happening due to the user's choice, not due to an error.
I'm also thinking to use a "may not" here since it's possible this is a re-deploy and manual GWs already exist that point to that function.

Reply:
we can also add a check if `self.status.api_gateway` is None or not